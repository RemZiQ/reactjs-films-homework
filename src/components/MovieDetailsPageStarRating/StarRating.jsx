import React from 'react';
import propTypes from 'prop-types';

import './starRating.scss';


const StarRating = ({
  title, duration, rating, genres,
}) => {
  const handling = (e) => {
    e.preventDefault();
  };
  const countOfStars = Math.round(rating);
  const listGenres = genres
    .map((item, index) => <li key={index}><button type="submit" onClick={handling}>{item}</button></li>);
  const stars = new Array(countOfStars).fill(1).map((item, index) => <li key={index}><i className="fas fa-star" /></li>);
  return (
    <div className="starRating_container">
      <h1>{title}</h1>
      <div className="starRating__genres_container">
        <div className="starRating__genres">
          <ul>
            {listGenres}
          </ul>
        </div>
        <div className="starRating__duration">
          <span>{duration}</span>
        </div>
        <div className="starRating__stars_container">
          <div className="starRating__stars">
            <ul>
              {stars}
            </ul>
          </div>
          <div className="starRating__rating">{rating}</div>
        </div>
      </div>
    </div>
  );
};


StarRating.propTypes = {
  title: propTypes.string,
  duration: propTypes.string,
  rating: propTypes.number,
  genres: propTypes.oneOfType([
    propTypes.arrayOf(propTypes.number),
    propTypes.arrayOf(propTypes.string),
  ]),
};

StarRating.defaultProps = {
  title: 'Sorry, no title here',
  duration: 'Sorry, no duration here',
  rating: 1,
  genres: [99],
};
export default StarRating;
