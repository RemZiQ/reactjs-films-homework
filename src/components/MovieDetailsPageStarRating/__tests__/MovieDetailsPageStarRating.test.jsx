import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
// import { Provider } from 'react-redux';
// import { MovieDetailsPageFooter } from './MovieDetailsPageFooter';
import { StarRating } from '../StarRating';


it('StarRating renders well', () => {
  const renderer = new ShallowRenderer();
  const result = renderer.render(<StarRating />);
  expect(result).toMatchSnapshot();
});

// it('renders well2222', () => {
//   const renderer = new ShallowRenderer();
//   const result = renderer.render(<MovieDetailsPageFooter store={{}} />);
//   expect(result).toMatchSnapshot();
// });

// it('renders well', () => {
//   const renderer = new ShallowRenderer();
//   const result = renderer.render(<MovieDetailsPageFooter store={null} />);
//   expect(result).toMatchSnapshot();
// });
