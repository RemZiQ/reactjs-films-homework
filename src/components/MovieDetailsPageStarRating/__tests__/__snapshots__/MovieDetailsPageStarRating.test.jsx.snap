// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StarRating renders well 1`] = `
<div
  className="starRating_container"
>
  <h1>
    Sorry, no title here
  </h1>
  <div
    className="starRating__genres_container"
  >
    <div
      className="starRating__genres"
    >
      <ul>
        <li>
          <button
            type="submit"
          />
        </li>
      </ul>
    </div>
    <div
      className="starRating__duration"
    >
      <span>
        Sorry, no duration here
      </span>
    </div>
    <div
      className="starRating__stars_container"
    >
      <div
        className="starRating__stars"
      >
        <ul>
          <li>
            <i
              className="fas fa-star"
            />
          </li>
        </ul>
      </div>
      <div
        className="starRating__rating"
      >
        1
      </div>
    </div>
  </div>
</div>
`;
